src/System.IdentityModel.Tokens.Jwt/JwtPayload.cs

```diff
@@ -529,7 +529,14 @@ public virtual IEnumerable<Claim> Claims
                             claims.Add(new Claim(keyValuePair.Key, "false", ClaimValueTypes.Boolean, issuer, issuer));
                     }
                     else if (keyValuePair.Value != null)
-                        claims.Add(new Claim(keyValuePair.Key, keyValuePair.Value.ToString(), GetClaimValueType(keyValuePair.Value), issuer, issuer));
+                    {
+                        var value = keyValuePair.Value;
+                        var claimValueType = GetClaimValueType(value);
+                        if (value is IConvertible convertible)
```

Comment:
IConvertible documentation indicates the interface is not CLS-Compliant.
Is that a problem? Currently, System.Identitymodel.Tokens.Jwt is CLS-Compliant.

https://learn.microsoft.com/en-us/dotnet/api/system.iconvertible?view=net-8.0